src/PostgREST/PgQuery.hs

```diff
@@ -241,7 +241,7 @@ pgFmtJsonbPath :: T.Text -> T.Text
 pgFmtJsonbPath p =
   pgFmtJsonbPath' $ fromMaybe (ColIdentifier p) (parseJsonbPath p)
   where
-    pgFmtJsonbPath' (ColIdentifier i) = pgFmtIdent i
+    pgFmtJsonbPath' (ColIdentifier i) = i
```

Comment:
I added another commit that rewrites some calls to whereT to make sure we always qualify and quote identifiers (the DELETE and PUT were missing). I believe now the PR is injection proof.

Another approach would be to keep the responsability to quote everything inside the PgQuery module, to do this I would need to add the qualified table as a parameter to the ```whereT``` and ```wherePred``` functions.

If you prefer the second approach let me know and I'll change the code ;)